pkg/machines/getOSList.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/python3
+
+import gi
+gi.require_version('Libosinfo', '1.0')
+from gi.repository import Libosinfo
+import sys
+
+
+loader = Libosinfo.Loader()
+loader.process_default_path()
+db = loader.get_db()
+
+oses = db.get_os_list()
+for i in range(oses.get_length()):
+    os = oses.get_nth(i)
+
+    osId = os.get_id()
+    if not osId:
+        osId = ""
+
+    osShortId = os.get_short_id()
+    if not osShortId:
+        osShortId = ""
+
+    osName = os.get_name()
+    if not osName:
+        osName = ""
+
+    osVersion = os.get_version()
+    if not osVersion:
+        osVersion = ""
+
+    osFamily = os.get_family()
+    if not osFamily:
+        osFamily = ""
+
+    osVendor = os.get_vendor()
+    if not osVendor:
+        osVendor = ""
+
+    osReleaseDate = os.get_release_date_string()
+    if not osReleaseDate:
+        osReleaseDate = ""
+
+    osEOLDate = os.get_eol_date_string()
+    if not osEOLDate:
+        osEOLDate = ""
+
+    osCodename = os.get_codename()
+    if not osCodename:
+        osCodename = ""
+
+    print("%s|%s|%s|%s|%s|%s|%s|%s|%s" %
+          (osId, osShortId, osName, osVersion, osFamily, osVendor,
+           osReleaseDate, osEOLDate, osCodename))
```

Comment:
Wouldn't it make more sense to print JSON here, so that it's less error prone to read from the JS side?

Reply:
It would. But, sincerely. I'd really prefer to **not** do this (and then having to change the code dealing with the result) in this patch.

This change could be done together with more improvements (as relying on Libosinfo for recommended resources/devices/machine type) later on, if you don't mind.

Reply:
Sure, it was just a suggestion, not a veto.